dll/cpl/sysdm/smbios.c

```diff
@@ -343,6 +435,13 @@ void AppendSystemFamily(PWSTR pBuf, SIZE_T cchBuf, PCHAR * DmiStrings, PWSTR dev
     }
 }
 
+static
+BOOL ContainsNonAscii(PCHAR DmiString)
+{
+    PCHAR p = strpbrk(DmiString, "\x01\x02\x03\x04\x05\x06\x07\x08\x09\x0A\x0B\x0C\x0D\x0E\x0F\x10\x11\x12\x13\x14\x15\x16\x17\x18\x19\x1A\x1B\x1C\x1D\x1E\x1F\xFF");
```

Comment:
What about everything in the range of 0x80 - 0xff ?

Reply:
If strpbrk performs a loop to check the characters, it may actually be more maintenable/readable to write a loop and just do the `ch >= 0x20 && ch <= 0x7e` test for each character in the string, instead of explicitly enumerating all the characters that have to be excluded.

Reply:
Fixed.